keras_cv/models/models_test.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Integration tests for KerasCV models."""
+
+import tensorflow as tf
+from absl.testing import parameterized
+from tensorflow.keras import backend
+
+from keras_cv.models import densenet
+
+MODEL_LIST = [
+    (densenet.DenseNet121, 1024),
+    (densenet.DenseNet169, 1664),
+    (densenet.DenseNet201, 1920),
+]
+
+
+class ApplicationsTest(tf.test.TestCase, parameterized.TestCase):
+    def assertShapeEqual(self, shape1, shape2):
+        if len(shape1) != len(shape2):
+            raise AssertionError(
+                "Shapes are different rank: %s vs %s" % (shape1, shape2)
+            )
+        for v1, v2 in zip(shape1, shape2):
+            if v1 != v2:
+                raise AssertionError("Shapes differ: %s vs %s" % (shape1, shape2))
+
+    @parameterized.parameters(*MODEL_LIST)
+    def test_application_base(self, app, _):
+        # Can be instantiated with default arguments
+        model = app(
+            input_shape=(224, 224, 3), include_preprocessing=False, weights=None
+        )
+        # Can be serialized and deserialized
+        config = model.get_config()
+        reconstructed_model = model.__class__.from_config(config)
+        self.assertEqual(len(model.weights), len(reconstructed_model.weights))
+        backend.clear_session()
```

Comment:
Not sure why we need the clear_session here.

Reply:
I imagine memory usage on CI.